File: app/src/main/java/com/orgzly/android/ui/dialogs/WhatsNewDialog.java
package com.orgzly.android.ui.dialogs;

import android.annotation.SuppressLint;
import android.app.AlertDialog;
import android.content.Context;
import android.text.Html;
import android.text.Spanned;
import android.text.method.LinkMovementMethod;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.TextView;

import com.orgzly.R;
import com.orgzly.android.util.MiscUtils;

public class WhatsNewDialog {
    /**
     * Display dialog with changes.
     */
    public static AlertDialog create(Context context) {
        @SuppressLint("InflateParams") final View layoutView =
                ((LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE))
                        .inflate(R.layout.dialog_whats_new, null, false);

        TextView view = ((TextView) layoutView.findViewById(R.id.dialog_whats_new_intro));
        view.setText(MiscUtils.fromHtml(context.getString(R.string.whats_new_intro)));
        view.setMovementMethod(LinkMovementMethod.getInstance());

        return new AlertDialog.Builder(context)
                .setTitle(R.string.whats_new_title)
                .setPositiveButton(R.string.ok, null)
                .setView(layoutView)
                .create();
    }
}

File: app/src/main/java/com/orgzly/android/ui/dialogs/WhatsNewDialog.kt
package com.orgzly.android.ui.dialogs

import android.content.Context
import android.text.method.LinkMovementMethod
import android.widget.TextView
import androidx.appcompat.app.AlertDialog
import com.google.android.material.dialog.MaterialAlertDialogBuilder
import com.orgzly.R
import com.orgzly.android.ui.util.getLayoutInflater
import com.orgzly.android.util.MiscUtils

object WhatsNewDialog {
    /**
     * Display dialog with changes.
     */
    fun create(context: Context): AlertDialog {
        val layoutView = context.getLayoutInflater().inflate(R.layout.dialog_whats_new, null, false)

        layoutView.findViewById<TextView>(R.id.dialog_whats_new_intro).apply {
            text = MiscUtils.fromHtml(context.getString(R.string.whats_new_intro))
            movementMethod = LinkMovementMethod.getInstance()
        }

        layoutView.findViewById<TextView>(R.id.dialog_whats_new_outro).apply {
            text = MiscUtils.fromHtml(context.getString(R.string.whats_new_outro))
            movementMethod = LinkMovementMethod.getInstance()
        }

        return MaterialAlertDialogBuilder(context)
            .setTitle(R.string.whats_new_title)
            .setPositiveButton(R.string.ok, null)
            .setView(layoutView)
            .create()
    }
}